src/main/io/gps.c

```diff
@@ -969,6 +1470,31 @@ static void gpsNewData(uint16_t c)
         return;
     }
 
+    gpsPackageCounter += 1;
+    if (millis() - gpsPackageCounterTime >= 1000) {
+        DEBUG_SET(DEBUG_GPS_UNIT_CONNECTION, 1, gpsPackageCounter);
+
+        // reconfigure update rate if not matching
+        if (
+            gpsData.state == GPS_STATE_RECEIVING_DATA
+        )
+        {
+            if (gpsPackageCounter > gpsData.updateRate - 2) {
+                gpsPackageReachedUpdateRate = true;
+            }
+            if (
+                gpsPackageReachedUpdateRate &&
+                gpsData.updateRate > gpsPackageCounter + 2
+            )
+            {
+                gpsSetState(GPS_STATE_CONFIGURE);
```

Comment:
this will trigger if few packets are lost. Is that intentional?

Reply:
it does seem a bit aggressive... at higher baud rates, random packet dropout can definitely be a problem.  Whether the threshold is too tight we need to keep an eye on.